Language/Haskell/GhcMod/ImportedFrom.hs

```diff
@@ -0,0 +1,826 @@
+
+        x <- runCmd rp "stack" opts
+
+        -- gmLog GmDebug "" $ strDoc $ "stackGhcPkgFindModule stdout: " ++ show output
+        -- gmLog GmDebug "" $ strDoc $ "stackGhcPkgFindModule stderr: " ++ show err
+        return $ case x of
+                    Just x' -> join $ (Safe.lastMay . words) <$> (Safe.lastMay . lines) x'
+                    Nothing -> Nothing
+
+ghcPkgHaddockUrl
+    :: forall m. (GmLog m, GmOut m, MonadIO m)
+    => FilePath
+    -> (FilePath -> [String] -> String -> IO String)
+    -> [GhcPkgDb]
+    -> String
+    -> m (Maybe String)
+ghcPkgHaddockUrl ghcPkg readProc pkgDbStack p = do
+    gmLog GmDebug "ghcPkgHaddockUrl" $ strDoc p
+
+    let p' = case splitOn "@" p of
+                [p0, _] -> p0
+                _       -> p
+
+    hout <- liftIO $ readProc ghcPkg (toDocDirOpts p' pkgDbStack) ""
+    return $ Safe.lastMay $ words $ reverse . dropWhile (== '\n') . reverse $ hout
+  where
+    -- This fails unless we have --global and --user, unlike
+    -- pkgDoc elsewhere in ghc-mod.
+    toDocDirOpts pkg dbs = ["field", pkg, "haddock-html", "--global", "--user"] ++ ghcPkgDbStackOpts dbs
+
+ghcPkgHaddockInterface
+    :: forall (m :: * -> *).  MonadIO m
+    => FilePath
+    -> (FilePath -> [String] -> String -> IO String)
+    -> [GhcPkgDb]
+    -> String
+    -> m (Maybe String)
+ghcPkgHaddockInterface ghcPkg readProc pkgDbStack p = do
+    hout <- liftIO $ readProc ghcPkg (toHaskellInterfaces p pkgDbStack) ""
+    return $ Safe.lastMay $ words $ reverse . dropWhile (== '\n') . reverse $ hout
+  where
+    toHaskellInterfaces pkg dbs = ["field", pkg, "haddock-interfaces", "--global", "--user"] ++ ghcPkgDbStackOpts dbs
+
+getVisibleExports
+    :: forall m. (GhcMonad m, MonadIO m, GmOut m, GmLog m)
+    => (String -> m (Maybe String))
+    -> String
+    -> m (Maybe (M.Map String [String]))
+getVisibleExports getHaddockInterfaces p = do
+    gmLog GmDebug "getVisibleExports" $ strDoc p
+
+    let p' = case splitOn "@" p of
+                [p0, _] -> p0
+                _       -> p
+
+    haddockInterfaceFile <- getHaddockInterfaces p'
+
+    case haddockInterfaceFile of
+        Just hi -> getVisibleExports' hi
+        Nothing -> return Nothing
+
+  where
+
+    getVisibleExports' :: forall m. (GhcMonad m, MonadIO m)
+                       => FilePath
+                       -> m (Maybe (M.Map String [String]))
+    getVisibleExports' ifile = do
+        iface <- Haddock.readInterfaceFile nameCacheFromGhc ifile
+
+        dflags <- GHC.getSessionDynFlags
+
+        case iface of
+            Left _          -> throw $ GMEMissingHaddockInterface ifile
+            Right iface'    -> do let m  = map (\ii -> (Haddock.instMod ii, Haddock.instVisibleExports ii)) $ Haddock.ifInstalledIfaces iface' :: [(Module, [Name])]
+                                      m' = map (\(mname, names) -> (showSDoc dflags $ ppr mname, map (showSDoc dflags . ppr) names)) m         :: [(String, [String])]
```

Comment:
I did point out the 80 char thing before (https://github.com/DanielG/ghc-mod/pull/810#discussion-diff-70889961) that applies everywhere in case that wasn't clear.